server/sonar-server/src/main/java/org/sonar/server/issue/ws/AssignAction.java

```diff
@@ -121,8 +120,9 @@ private void assign(String issueKey, @Nullable String assignee) {
       }
       IssueChangeContext context = IssueChangeContext.createUser(new Date(system2.now()), userSession.getLogin());
       if (issueFieldsSetter.assign(issue, user, context)) {
-        issueUpdater.saveIssue(dbSession, issue, context, null);
+        return issueUpdater.saveIssueAndPreloadSearchResponseData(dbSession, issue, context, null);
```

Comment:
almost true.

The login in IssueChangeContext is the login of the caller of the WS.
The users which are loaded from DB to build the response is the assignee of the issue. Apparently, we don't care about the current user, except for issues/search (see https://github.com/SonarSource/sonarqube/blob/master/server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java#L415-L419).
In addition, the request on table USERS by id is fast (table is small compared to projects or issues).

In the current state, the next winner is removing the SQL request on table PROJECTS which retrieves all modules. This request is commented as "here for backward compatibility" but it is costly (50ms locally) (this one: https://github.com/SonarSource/sonarqube/blob/master/server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchResponseLoader.java#L123).